### PR TITLE
connector: Doc the errors of NewAWSReader

### PR DESCRIPTION
**Pull request by ifraixedes**
When using this service and wanted to actually know which errors can be
returned by the NewAWSReader, the implementation of it had to be
checked. For avoiding of having to do it each time by whoever needed it,
the errors that the NewAWSReader have been documented.

This PR has documented some internal functions because it was the way to
track from the bottom-up which errors the NewAWSReader can return, it
was not necessary for the end purpose, but having such docs are more
helpful than harmful.

### CODE REVIEW

File: connector.go [hunk 1]
```diff
@@ -173,6 +173,13 @@ type AWSReader interface {
 //
 // The connections are not all established while instancing, but the various sessions are, this way connections are only
 // made for services that are called, otherwise only the sessions remain.
+//
+// An error is returned if any of the needed AWS request for creating the reader returns an AWS error, in such case it
+// will have any of the common error codes (see below) or EmptyStaticCreds code or a go standard error in case that no
+// regions are matched with the ones available, at the time, in AWS.
+// See:
+//	* https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html#CommonErrors
+//  * https://docs.aws.amazon.com/STS/latest/APIReference/CommonErrors.html
```
Comment (xescugc) [inline connector.go]: Remove the space or add it on the first one hehe

**Review by xescugc — APPROVED** [review]
Just an stupid thing, if not it's ok haha

File: connector.go (hunk 1 above)
Comment (ifraixedes) [reply connector.go]: I couldn't see it visually in my editor, basically, the first one was a tab and the second 2 spaces; I wanted to have 2 spaces, so I updated the first one.

**Comment by xescugc** [comment]
You can merge it :+1: (it's approved already hehe)

**Pull request merged**